bebraven/platform
# Migrate CourseContentHitories to CustomContentVersions

**Task**: https://app.asana.com/0/1174274412967132/1186960110311121

### Description

- This is the new data model for project submissions: https://dbdiagram.io/d/5f6e41177da1ea736e2f6328
- This PR specifically migrates and renames `CourseContentHistory` to `CustomContentVersion`
- This PR shouldn't have any user-facing impact, since we only interact with this object behind-the-scene when Publishing from Content Editor to Portal, or inserting a project through Canvas's LTI assignment placement.

### Motivation and Context

For expedience, we put project submission logic in `CourseContentHistories`. The reasoning behind this is because a Fellow's project submission is always tied to a particular version of the project (e.g., a `CourseContentHistory` object). 

Overloading the `CourseContentHistoriesController` is getting confusing and making it difficult for us to implement things like showing multiple project submissions per fellow correctly. 

The first step in implementing the remaining project submission user stories is to refactor the code. 

### Test

- Run `rspec`
- Go to `platformweb/course_contents/#/versions` and make sure versions still render
<img width="1792" alt="Screen Shot 2020-09-29 at 4 10 47 PM" src="https://user-images.githubusercontent.com/62911934/94625814-8a025980-026e-11eb-9866-b71e6e8f9ce0.png">

- Go to Canvas and create a new Assignment with a project saved in the Content Editor, have student submit project
<img width="1792" alt="Screen Shot 2020-09-29 at 4 11 04 PM" src="https://user-images.githubusercontent.com/62911934/94625838-97b7df00-026e-11eb-8122-c069037430c7.png">

- Make sure the submission still renders in the SpeedGrader
<img width="1792" alt="Screen Shot 2020-09-29 at 4 10 54 PM" src="https://user-images.githubusercontent.com/62911934/94625821-8cfd4a00-026e-11eb-9b44-2033dccb4498.png">

### Types of changes




### Checklist:
